runtime/criusupport/criusupport.c

```diff
@@ -161,9 +148,7 @@ getNativeString(J9VMThread *currentThread, j9object_t javaString, char **nativeS
 	if (requiredConvertedStringSize > STRING_BUFFER_SIZE) {
 		*nativeString = (char*)j9mem_allocate_memory(requiredConvertedStringSize, OMRMEM_CATEGORY_VM);
```

Comment:
This makes the allocated buffer the callers problem, even if the conversion fails.  Better to only write to `*nativeString` when the api has succeed so that all freeing of buffers in error cases is localized in this function.

Basically, only write the buffer to `*nativeString` on success

Reply:
Are you saying to use a local pointer in place of `*nativeString` and set `*nativeString` to that pointer upon success?

Reply:
You pass in a buffer so the code should point the local pointer to that buffer initially and in the case we have to allocate, only pass the allocated pointer back on success.  It's simpler if the caller only takes ownership of the allocated memory on success.

This is the kind of pattern your looking for:
```
char *localNativeString = *nativeString;
....
localNativeString = j9mem_allocate_memory(requiredConvertedStringSize, OMRMEM_CATEGORY_VM);
....
/* check if we're using the passed in buffer or not */
if (localNativeString != *nativeString) {
   if (success) {
      /* pass the locally allocated string to the caller.  They own the allocation now */
      *nativeString = localNativeString;
   } else {
      free(localNativeString);
      localNativeString = NULL;
   }
}
```